dvc/output.py

```diff
@@ -709,20 +709,21 @@ def verify_metric(self):
                 f"verify metric is not supported for {self.protocol}"
             )
 
-        if not self.metric or self.plot:
+        if not (self.metric or self.plot):
```

Comment:
This change is not related to the rest of the PR. The original line looked like a mistake (I make this kind of mistake all the time) since the code below (especially `name = "metrics" if self.metric else "plot"`) implied that plots still needed to be checked. But now the test `test_ignore_parsing_error` fails when calling `run_copy_metrics`. It's hard to tell, but it looks like the test is meant to be testing that errors are ignored in `show`, not `run`. So should I update the test to not fail on `run_copy_metrics`? Or should I remove checks for plots in this method?

Reply:
This was not a bug. The line setting name to `plot` was left over from a point when plots did need to be checked, but that is no longer the case, plots are now permitted to be non-text files.

Reply:
Yep, this check blocks later potential exceptions that might occur, for:
1. non text files - as Peter mentioned we do support binaries for some time now
2. dirs - plots can be dirs

"plots" mentions in strings seems to be obsolete logic that haven't been caught when introducing support for images/dirs. We should probably fix that, but it should be a separate PR. @alexmojaki can I ask you to create issue for that?

Reply:
Is it possible for something to be both a metric and a plot?

Reply:
In the usual use case, no. Though we have (now deprecated, but still supported till 3.0) `live` output which is using logic that was a mix of both. Why?

Reply:
I was wondering what the cleaned version of this code would look like. Keeping `if not self.metric or self.plot:` as is instead of just `if not self.metric:` seems to imply that `self.metric and self.plot` might be true, in which case checks would be skipped.

Reply:
Huh, this is a very valid point. I don't think there is a use case where `self.metric==True and self.plots==True`. I guess we could simplify this check.